extensions/in-memory/contractdefinition-store-memory/src/test/java/org/eclipse/dataspaceconnector/contractdefinition/store/memory/InMemoryContractDefinitionStoreTest.java

```diff
@@ -43,10 +44,16 @@ void verifyStore() {
         store.save(List.of(definition2));
         assertThat(store.findAll()).contains(definition1);
 
-        store.delete(definition1.getId());
+        var deletedDefinition = store.deleteById(definition1.getId());
+        assertThat(deletedDefinition).isEqualTo(definition1);
         assertThat(store.findAll()).doesNotContain(definition1);
     }
 
+    @Test
+    void deleteById_whenContractDefinitionMissing_returnsNull() {
+        assertThat(store.deleteById(UUID.randomUUID().toString())).isNull();
```

Comment:
Relates to previous comment regarding null vs. error; if this is the desired functionality then this is good